R/create_dummy_data_com.R

```diff
@@ -67,14 +67,15 @@ create_dummy_data_com <- function(df_type = "student") {
                                                            40,  #set day to 40 (DNE)
                                                            .data$BirthDay)))) %>%
     dplyr::select(-c(.data$BirthYear,.data$BirthMonth,.data$BirthDay)) %>%
-    #NOTE: this is not an exact Age calculuation -- it doesn't handle leap years.
+    #NOTE: this is not an exact Age calculation -- it doesn't handle leap years.
     # I figured this is good enough for testing the Upload logic
     # To compute age including leap years, use logic that looks at year, then month, than day relative
     # to the arbitrary completions date used below.
     dplyr::mutate(Age = (lubridate::ymd('2018-05-16') - .data$Birthdate)/lubridate::dyears(1)) %>%
     #adding more detailed sex info for 2022+ reporting years
-    dplyr::mutate(GenderDetail = dplyr::case_when(.data$StudentId < 103 ~ 3, #unknown
-                                               .data$StudentId < 105 ~ 4, #another
+    dplyr::mutate(GenderDetail = dplyr::case_when(.data$StudentId < 103 ~ 3, #unknown UG (3)
+                                               .data$StudentId < 105 ~ 4, #another UG (2)
+                                               .data$StudentId > 154 ~ 4, #another GR (5)
```

Comment:
I don't understand the logic on the GenderDetail case_when() in the dummy data. What happens to StudentId's between 104 and 154? Why are the first two case statements < 103 and < 105?

Reply:
I believe we set the GenderDetail defaults to be the same as Sex above, and the student ID numbers range from 101-155, so this is setting the low numbers 101-103 as UNK UG level, 104-105 as ANO for UG and then the top numbers 154-155 as ANO for GR level.  We ignore the middle because it stays as M (1) F(2).